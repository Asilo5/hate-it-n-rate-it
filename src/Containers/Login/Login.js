import React, { Component } from 'react';
import { fetchUser, fetchRatings } from '../../utils/apiCalls';
import { Redirect } from 'react-router-dom';
import { setUser, hasError, setUserRatings } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import './Login.scss';

export class Login extends Component {
  constructor() {
    super();
    this.state = { 
      email: '',
      password: '',
      userFound: false
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleSubmit = () => {
    const { email, password } = this.state;
    fetchUser(email, password) 
      .then(data => {
        this.props.setUser(data);
        this.handleUserRatings(data.user.id);
        this.setState({ 
          userFound: true, 
          email: '',
          password: '' 
        });
      })
      .catch(() => {
        this.setState({ 
          userFound: false,
          email: '',
          password: '' 
        });
        this.props.hasError('Email or password are incorrect, please try again!');
      });
    }
    
    handleUserRatings = (userId) => {
    fetchRatings(userId)
      .then(data => this.props.setUserRatings(data))
  }

  render() {
    const { error } = this.props;
    const { userFound } = this.state;

    if (userFound) {
      return <Redirect to='/' />
    }

    return (
      <section>
        <form>
          <label>Email:</label>
          <input 
            name='email' 
            value={this.state.email} 
            onChange={(e) => this.handleChange(e)}
            />
          <label>Password:</label>
          <input 
            type='password'
            name='password' 
            value={this.state.password} 
            onChange={(e) => this.handleChange(e)}
            />
          <p className='error'>{error}</p>
            <div>
              <button
                className='login_button'
                type='button'
                onClick={this.handleSubmit}
              >LOGIN</button>
            </div>
        </form>
      </section>
    )
  }
}

export const  mapStateToProps = ({ user, error }) => ({
  user,
  error 
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
  setUser: user => dispatch(setUser(user)),
  hasError: error => dispatch(hasError(error)),
  setUserRatings: ratedMovies => dispatch(setUserRatings(ratedMovies))
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(Login);
