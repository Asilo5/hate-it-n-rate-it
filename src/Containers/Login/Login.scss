form {
    padding: 20px;
    display: flex;
    flex-direction: column;
    width: 30% auto;
    width: 30%;
    margin: auto;
    position: relative;
    top: 16em;
    label {
        font-size: 2em;
        font-weight: bold;
    }
    input {
        font-size: 25px;
        border: none;
        border-bottom: 2px solid black;
        margin: 5px 0;
        caret-color: red;
    }
    .login_button {
        text-decoration: none;
        padding: 8px;
        margin: 12px auto;
        border: 2px solid black;
        width: 30%;
        text-align: center;
        font-size: 2em;
        display: block;
        border-radius: 5px;
        color: black;
        font-weight: bold;
        &:hover {
            background-color: black;
            color: white;
        }
    }
    .error {
        color: red;
        font-size: 23px;
        text-align: center;
    }

    .welcome-msg {
        @extend .error;
        color: green;
    }
}