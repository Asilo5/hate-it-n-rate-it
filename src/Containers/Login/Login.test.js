import React from 'react';
import { Login } from './Login';
import { shallow } from 'enzyme';

describe('Login', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  })
})
