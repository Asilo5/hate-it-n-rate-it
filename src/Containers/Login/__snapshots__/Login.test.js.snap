// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
<section>
  <form>
    <input
      name="email"
      onChange={[Function]}
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <NavLink
      className="login_button"
      onClick={[Function]}
      to="/"
      type="button"
    >
      LOGIN 
    </NavLink>
  </form>
</section>
`;
